test/lint/lint-filenames.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Make sure only lowercase alphanumerics (a-z0-9), underscores (_) and
+# hyphens (-) are used in source code filenames.
+
+EXIT_CODE=0
+OUTPUT=$(git ls-files -- "*.cpp" "*.h" | grep -vE '^[a-z0-9_./-]+$' | grep -vE 'src/(secp256k1|univalue)/')
```

Comment:
Just noting that this regex doesn't work for me:

```
echo 'A'|grep -E '[a-z]'

Reply:
Oh, another case of locale dependence! `echo 'A' | LANG=C grep -E '[a-z]'` works as expected, right?

Now updated with `LANG=C`.

Reply:
Works now.


```sh
$ set | grep -E '^(LANG|LC_)' && bash --version | head -1 && shopt | grep globasciiranges && lsb_release -d
LANG=en_US.UTF-8
GNU bash, version 4.4.19(1)-release (x86_64-redhat-linux-gnu)
globasciiranges	off
Description:	Fedora release 28 (Twenty Eight)